File: frontend/src/services/recycleBinService.ts
import apiClient from './apiClient'
import type { ApiResponse, PagedResult } from '../types/common'

export interface DeletedAccountResponse {
  id: number
  websiteId: number
  websiteDomain: string
  websiteDisplayName?: string
  username: string
  password: string
  notes?: string
  tags?: string
  createdAt: string
  updatedAt: string
  deletedAt?: string
}

class RecycleBinService {
  private readonly baseUrl = '/api/recycle-bin'

  async getDeletedAccounts(
    pageNumber: number = 1,
    pageSize: number = 10,
    websiteId?: number
  ): Promise<ApiResponse<PagedResult<DeletedAccountResponse>>> {
    const params = new URLSearchParams({
      pageNumber: pageNumber.toString(),
      pageSize: pageSize.toString(),
    })

    if (websiteId) {
      params.append('websiteId', websiteId.toString())
    }

    return await apiClient.get<PagedResult<DeletedAccountResponse>>(
      `${this.baseUrl}?${params.toString()}`
    )
  }

  async restoreAccount(id: number): Promise<ApiResponse<{ message: string }>> {
    return await apiClient.post<{ message: string }>(
      `${this.baseUrl}/${id}/restore`
    )
  }

  async permanentlyDeleteAccount(
    id: number
  ): Promise<ApiResponse<{ message: string }>> {
    return await apiClient.delete<{ message: string }>(`${this.baseUrl}/${id}`)
  }

  async emptyRecycleBin(
    websiteId?: number
  ): Promise<ApiResponse<{ message: string }>> {
    const params = websiteId ? `?websiteId=${websiteId}` : ''
    return await apiClient.delete<{ message: string }>(
      `${this.baseUrl}${params}`
    )
  }
}

export const recycleBinService = new RecycleBinService()

File: backend/src/AccountBox.Api/Services/RecycleBinService.cs
using AccountBox.Core.Models;
using AccountBox.Core.Models.RecycleBin;
using AccountBox.Data.Repositories;

namespace AccountBox.Api.Services;

/// <summary>
/// 回收站业务服务
/// 管理已删除账号的查看、恢复和永久删除（明文存储模式）
/// </summary>
public class RecycleBinService
{
    private readonly IAccountRepository _accountRepository;

    public RecycleBinService(IAccountRepository accountRepository)
    {
        _accountRepository = accountRepository ?? throw new ArgumentNullException(nameof(accountRepository));
    }

    /// <summary>
    /// 获取回收站中的分页账号列表
    /// </summary>
    public async Task<PagedResult<DeletedAccountResponse>> GetDeletedAccountsAsync(
        int pageNumber,
        int pageSize,
        int? websiteId)
    {
        var (items, totalCount) = await _accountRepository.GetDeletedPagedAsync(pageNumber, pageSize, websiteId);

        var deletedAccountResponses = items.Select(MapToDeletedResponse).ToList();

        return new PagedResult<DeletedAccountResponse>
        {
            Items = deletedAccountResponses,
            TotalCount = totalCount,
            PageNumber = pageNumber,
            PageSize = pageSize
        };
    }

    /// <summary>
    /// 恢复账号（从回收站恢复到活跃状态）
    /// </summary>
    public async Task RestoreAccountAsync(int accountId)
    {
        try
        {
            await _accountRepository.RestoreAsync(accountId);
        }
        catch (KeyNotFoundException)
        {
            throw new KeyNotFoundException($"Account with ID {accountId} not found in recycle bin");
        }
        catch (InvalidOperationException ex)
        {
            throw new InvalidOperationException($"Cannot restore account: {ex.Message}");
        }
    }

    /// <summary>
    /// 永久删除账号（从数据库物理删除）
    /// </summary>
    public async Task PermanentlyDeleteAccountAsync(int accountId)
    {
        try
        {
            await _accountRepository.PermanentlyDeleteAsync(accountId);
        }
        catch (KeyNotFoundException)
        {
            throw new KeyNotFoundException($"Account with ID {accountId} not found");
        }
    }

    /// <summary>
    /// 清空回收站（永久删除所有已删除的账号）
    /// </summary>
    public async Task EmptyRecycleBinAsync(int? websiteId = null)
    {
        await _accountRepository.EmptyRecycleBinAsync(websiteId);
    }

    /// <summary>
    /// 将 Account 实体映射到 DeletedAccountResponse DTO
    /// </summary>
    private DeletedAccountResponse MapToDeletedResponse(Data.Entities.Account account)
    {
        return new DeletedAccountResponse
        {
            Id = account.Id,
            WebsiteId = account.WebsiteId,
            WebsiteDomain = account.Website?.Domain ?? string.Empty,
            WebsiteDisplayName = account.Website?.DisplayName,
            Username = account.Username,
            Password = account.Password, // 直接返回明文密码
            Notes = account.Notes,
            Tags = account.Tags,
            CreatedAt = account.CreatedAt,
            UpdatedAt = account.UpdatedAt,
            DeletedAt = account.DeletedAt
        };
    }
}
